portfolio/src/main/java/com/google/sps/data/CommentRequest.java

```diff
@@ -24,11 +24,17 @@
   /** List of requested comments. */
   private List<Comment> comments;
 
-  /** Total number of comments. */
-  private int total;
+  /**
+   * Total number of comments found from the query. If no limit is set and all comments asked for
+   * are returned, total should be equal to the size of List<Comment> comments. Alternatively, if
+   * the limit is greater than the number of comments returned, then total should also be equal to
+   * the size of List<Comment> comments. Otherwise, only a subset of all the comments found will
+   * be returned to the client.
```

Comment:
From my understanding of your code, I think this is a little inaccurate/misleading.  Correct me if I'm wrong, but based on what I see below of how you create a CommentRequest, `totalComments` always contains the total number of comments, no?  (With the exception of if there are more comments than MAX_INTEGER, then it contains MAX_INTEGER.). So I don't see the need to include all of this conditional explaining in the comment.  I think instead you could just say something like "Total number of comments existing in Datastore" (to make clear that "comments" doesn't refer to your variable above.  Instead, I would include what you're alluding to here in the comment above for the "comments" variable.  

And as a nit, I would say this comment is perhaps a little verbose.  If you're using intuitive variable names, which you now are :) you can avoid having to explain so much in the comment because the reader will be able to intuit more from the code.  So you could probably condense this by saying something like "A list that contains a strict subset of the comments in Datastore if fewer than the total were requested, else all of the comments, with a maximum of MAX_INTEGER."

Reply:
Yep, you're right! I'll update the description then :)